debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java

```diff
@@ -62,7 +62,8 @@ public void processNotEmptyMessage(ByteBuffer buffer, ReplicationMessageProcesso
             // WAL2JSON may send empty changes that still have a txid. These events are from things like vacuum,
             // materialized view, DDL, etc. They still need to be processed for the heartbeat to fire.
             if (changes.isEmpty()) {
-                processor.process(null);
+                processor.process(new TransactionMessage(Operation.BEGIN, txId, commitTime));
```

Comment:
Why not the NOOP message here?

Reply:
@gunnarmorling Because it is different - NOOP is used in case of filtered event in TX. This is an empty TX is the valid logic is not emit NOOP but begin and end for the empty TX.